mempool/clist_mempool.go

```diff
@@ -542,23 +551,37 @@ func (mem *CListMempool) resCbFirstTime(
 func (mem *CListMempool) resCbRecheck(req *abci.Request, res *abci.Response) {
 	switch r := res.Value.(type) {
 	case *abci.Response_CheckTx:
+		tx := req.GetCheckTx().Tx
+		txHash := TxKey(tx)
+		e, ok := mem.txsMap.Load(txHash)
+		if !ok {
+			mem.logger.Debug(
+				"re-CheckTx transaction does not exist",
+				"expected", types.Tx(tx),
+			)
+			return
+		}
+
```

Comment:
I think that `logging` and `removeTx` are redundant. In addition, `removeTx`'s option has a bug in the current version. It should use `!mem.config.KeepInvalidTxsInCache` in `recCbRecheck`.

* https://github.com/tendermint/tendermint/blob/v0.34.19/mempool/clist_mempool.go#L477
* https://github.com/line/ostracon/blame/archived/apply_v0.33.9-1/mempool/clist_mempool.go#L465
* https://github.com/line/ostracon/blob/v1.0.1/mempool/clist_mempool.go#L550

How about the below?

```
		var postCheckErr error
		if r.CheckTx.Code == abci.CodeTypeOK {
			if mem.postCheck == nil {
				return
			}
			postCheckErr = mem.postCheck(tx, r.CheckTx)
			if postCheckErr == nil {
				return
			}
			r.CheckTx.MempoolError = postCheckErr.Error()
		}
		celem := e.(*clist.CElement)
		// Tx became invalidated due to newly committed block.
		mem.logger.Debug("tx is no longer valid", "tx", txID(tx), "res", r, "err", postCheckErr)
		// NOTE: we remove tx from the cache because it might be good later
		mem.removeTx(tx, celem, !mem.config.KeepInvalidTxsInCache)
```